src/main/scala/cromwell/engine/backend/local/LocalEngineFunctions.scala

```
@@ -96,9 +111,32 @@ class LocalEngineFunctions(cwd: Path, stdout: Path, stderr: Path) extends LocalE
       singleArgument <- extractSingleArgument(params)
       if singleArgument.wdlType.isInstanceOf[WdlMapType]
       tsvSerialized <- singleArgument.asInstanceOf[WdlMap].tsvSerialize
-      (path, writer) = FileUtil.tempFileAndWriter("map", cwd.toFile)
-      _ <- Try(writer.write(tsvSerialized))
-      _ <- Success(writer.close())
-    } yield WdlFile(path.toAbsolutePath.toString)
+      file = writeContent("map", tsvSerialized)
+    } yield file
+  }
+
+  override protected def write_object(params: Seq[Try[WdlValue]]): Try[WdlFile] = {
+    for {
+      singleArgument <- extractSingleArgument(params)
+      if singleArgument.wdlType == WdlObjectType
+      tsvSerialized <- singleArgument.asInstanceOf[WdlObject].tsvSerialize
+      file = writeContent("object", tsvSerialized)
+    } yield file
+  }
+
+  override protected def write_objects(params: Seq[Try[WdlValue]]): Try[WdlFile] = {
+    for {
+      singleArgument <- extractSingleArgument(params)
+      if singleArgument.wdlType.isAnArrayOf(WdlObjectType)
+      tsvSerialized <- singleArgument.asInstanceOf[WdlArray].tsvSerialize
+      file = writeContent("array", tsvSerialized)
+    } yield file
+  }
+
+  protected def writeContent(baseName: String, content: String) = {
+    val (path, writer) = FileUtil.tempFileAndWriter("array", cwd.toFile)
+    Try(writer.write(content))
```

Is this ignoring any error that might happen with `writer.write`?